app.js

```diff
@@ -131,14 +131,31 @@ submitButton.addEventListener("click", function () {
 });
 
 // Add event listener to new game button
-
 newGameButton.addEventListener("click", function () {
-  clearGrid();
-  rollThatDice();
-  emptyLeaderboard();
-  window.location.reload();
+  // Check if any squares are marked as occupied
+  const occupiedSquares = document.querySelectorAll(".occupied");
```

Comment:
const occupiedSquares = document.querySelectorAll(".block");

it's the only part I would change, the function is really great :)